docs/guides/golang.md

```diff
@@ -0,0 +1,251 @@
+that responds to a `CloudEvent` or simple HTTP. `CloudEvents` in Knative are transported over
+HTTP as a `POST` request, so in many ways, the two types of functions are very much the same.
+They each will listen and respond to incoming HTTP events.
+
+#### Function triggered by HTTP request
+
+When an incoming request is received, your function will be invoked with a standard
+Golang [Context](https://golang.org/pkg/context/) as the first parameter followed by
+two parameters: Golang's [http.ResponseWriter](https://golang.org/pkg/net/http/#ResponseWriter)
+and [http.Request](https://golang.org/pkg/net/http/#Request). 
+
+Then you can use standard Golang techniques to access the request (eg. read the body)
+and set a proper HTTP response of your function, as you can see on the following example:
+
+```go
+func Handle(ctx context.Context, res http.ResponseWriter, req *http.Request) {
+
+  // Read body
+  body, err := ioutil.ReadAll(req.Body)
+  defer req.Body.Close()
+  if err != nil {
+	http.Error(res, err.Error(), 500)
+	return
+  }
+
+  // Process body & function logic
+  // ...
+}
+```
+
+#### Function triggered by CloudEvent
+
+If the incoming request is a `CloudEvent`, the event is provided via
+[CloudEvents Golang SDK](https://cloudevents.github.io/sdk-go/) and it's `Event` type
+as a parameter. There's possibility to leverage Golang's
+[Context](https://golang.org/pkg/context/) as the optional parameter in the function contract,
+as you can see in the list of supported function signatures:
+
+```go
+Handle()
+Handle() error
+Handle(context.Context)
+Handle(context.Context) error
+Handle(cloudevents.Event)
+Handle(cloudevents.Event) error
+Handle(context.Context, cloudevents.Event)
+Handle(context.Context, cloudevents.Event) error
+Handle(cloudevents.Event) *cloudevents.Event
+Handle(cloudevents.Event) (*cloudevents.Event, error)
+Handle(context.Context, cloudevents.Event) *cloudevents.Event
+Handle(context.Context, cloudevents.Event) (*cloudevents.Event, error)
+```
+
+For example, a `CloudEvent` is received which contains a JSON string such as this in its data property, 
+
+```json
+{ 
+  "customerId": "0123456",
+  "productId": "6543210"
+}
+```
+
+to access this data, we need to define `Purchase` structure, which maps properties in `CloudEvents`
+data and retrive it from the incoming event:
+
+```go
+type Purchase struct {
+  CustomerId string `json:"customerId"`
+  ProductId  string `json:"productId"`
+}
+
+func Handle(ctx context.Context, event cloudevents.Event) err error {
+	  
+  purchase := &Purchase{}
+  if err = cloudevents.DataAs(purchase); err != nil {
+	fmt.Fprintf(os.Stderr, "failed to parse incoming CloudEvent %s\n", err)
+	return
+  }
+
+  // ...
+}
+```
+
+Or we can use Golang's `encoding/json` package to access the `CloudEvent` directly as
+a JSON in form of bytes array:
+
+```golang
+func Handle(ctx context.Context, event cloudevents.Event) {
+
+  bytes, err := json.Marshal(event)
```

Comment:
I think that you should call `event.Data()` when working with raw data (e.g. if data are *.jpeg). Am I right @slinkydeveloper ?

Reply:
This is what [CE Go SDK ](https://cloudevents.github.io/sdk-go/#serializedeserialize-a-cloudevent) mentions for marshalling to json.

Reply:
@zroubalik @matejvasek are we good to land this?

Reply:
I am, not sure what @matejvasek thinks

Reply:
I thinks this is all right.  I just don't see why I would transformed CE to JSON.